implementations/rust/ockam/ockam_app/src/projects/commands.rs

```diff
@@ -0,0 +1,40 @@
+use std::sync::Arc;
+
+use tauri::{async_runtime::RwLock, AppHandle, Manager, Runtime, State};
+use tracing::{debug, info};
+
+use ockam_api::cloud::project::Project;
+use ockam_command::util::api::CloudOpts;
+
+use super::State as ProjectState;
+use crate::app::AppState;
+
+type SyncState = Arc<RwLock<ProjectState>>;
+
+// At time of writing, tauri::command requires pub not pub(crate)
+
+#[tauri::command]
+pub async fn list_projects<R: Runtime>(app: AppHandle<R>) -> Result<Vec<Project>, String> {
+    let state: State<'_, SyncState> = app.state();
+    let reader = state.read().await;
+    Ok((*reader).clone())
+}
+
+#[tauri::command]
+pub async fn refresh_projects<R: Runtime>(app: AppHandle<R>) -> Result<(), String> {
+    info!("refreshing projects");
+    let state: State<'_, AppState> = app.state();
+    let node_manager_worker = state.node_manager_worker().await;
+    let projects = node_manager_worker
+        .list_projects(&state.context(), &CloudOpts::route())
+        .await
+        .map_err(|e| e.to_string())?;
+    debug!(?projects);
```

Comment:
Something to keep in mind, we would probably to update the CliState so that the nodes that we create can read that data from there:

1. To add/update the projects data
2. To update the node's projects data

We do this update [here](https://github.com/build-trust/ockam/blob/40409c92e1a0de1e00bfcb8e4d924466db6a9377/implementations/rust/ockam/ockam_app/src/enroll/enroll_user.rs#L133) for example.

Reply:
How does 277bea938234d719237c69c806f7f50d3093d5f0 look for this?

Reply:
Perfect!